### PR TITLE
Fix: Return Unprocessable instead of error for previously failed messages

### PR DESCRIPTION
**Pull request by mubarakcoded**
## Summary
Fixes crash when processing messages that previously failed by returning `MessageProcessingResult::Unprocessable` instead of throwing an error. This prevents application crashes when duplicate failed messages arrive from Nostr relays.

## Problem
After commit 652198d (DoS protection), when a message fails processing (e.g., epoch mismatch, decryption failure), it's marked as `ProcessedMessageState::Failed`. If the same message arrives again, MDK throws an error instead of returning a result, causing crashes in applications that don't wrap `processMessage()` in try-catch blocks.

This particularly affects:
- Android Kotlin bindings users
- Flutter applications
- Any app receiving batches of messages from relays

## Solution
Changed the error handling to return `Ok(MessageProcessingResult::Unprocessable { mls_group_id })` instead of `Err(Error::Message(...))` for previously failed messages with valid h-tags.

### Important caveat
**Previously failed messages with malformed/missing h-tags still return `Err`**. This is intentional based on reviewer feedback - we prefer explicit errors over magic zero values when we can't determine the group ID. This edge case is rare (malicious/buggy events arriving twice) and apps should handle errors appropriately.

## What changed
- `process_message` now returns `Ok(MessageProcessingResult::Unprocessable { mls_group_id })` for previously failed messages with valid h-tags
- Added `extract_group_id_from_failed_message` helper that returns `Option<GroupId>`
- When h-tag extraction fails, returns `Err` with "Message processing previously failed"
- Updated tests to cover both valid h-tag (returns `Unprocessable`) and malformed h-tag (returns `Err`) scenarios

## Security impact
- No changes to cryptographic primitives, key derivation, nonces, or key storage
- DoS-protection semantics preserved
- No new error messages that would leak sensitive identifiers

This PR prevents application crashes on re-delivery of messages that previously failed by treating them as non-fatal unprocessable messages when possible, aligning their handling with other unprocessable message cases and preserving DoS-protection semantics. It adds a private helper to extract a group id from the Nostr h-tag for failed messages and updates tests and the changelog to cover the new behavior and edge cases.

**What changed**:
- process_message now returns Ok(MessageProcessingResult::Unprocessable { mls_group_id }) for messages previously marked ProcessedMessageState::Failed when a valid h-tag can be extracted.  
- Added private helper MDK::extract_group_id_from_failed_message(&self, event: &Event) -> Option<GroupId> that attempts to parse the Nostr h-tag, hex-decode it to a nostr group id, prefer a stored MLS group id when available, and return None when extraction fails.  
- If h-tag extraction fails due to malformed or missing tags, the code still returns Err(Error::Message("Message processing previously failed" ...)) to avoid using magic zero identifiers.  
- Updated debug logging to report sanitized, non-fatal Unprocessable results and group-id extraction outcomes.  
- Updated crates/mdk-core/CHANGELOG.md to document the fix.  
- Affected crates: mdk-core (source + CHANGELOG); no source changes detected in mdk-sqlite-storage, mdk-memory-storage, mdk-storage-traits, or mdk-uniffi.

**Security impact**:
- No changes to cryptographic primitives, key derivation, nonces, or key storage.  
- No identity binding or credential validation changes beyond existing behavior.  
- No new error messages that leak sensitive identifiers; failure reasons remain sanitized.  
- No SQLCipher configuration or file permission changes.

**Protocol changes**:
- No MLS protocol (RFC 9420) implementation changes.  
- No Nostr/NIP semantic changes; only the handling/reporting of re-delivered failed Nostr-wrapped MLS messages was adjusted.

**API surface**:
- No breaking changes to public APIs, types, or traits; MessageProcessingResult::Unprocessable was already an existing variant and remains unchanged in signature.  
- The new helper is private; no new public APIs, storage schema changes, or UniFFI/FFI boundary changes.

**Testing**:
- Unit tests in crates/mdk-core were added/expanded to cover Unprocessable behavior and group-id extraction fallback paths, including valid stored group IDs, malformed/oversized/undersized h-tags, invalid hex decoding, and None fallback scenarios.

### CODE REVIEW

File: crates/mdk-core/src/messages.rs [hunk 1]
```diff
@@ -1720,6 +1720,39 @@ where
...
+            .and_then(|bytes| bytes.try_into().ok())
+            .and_then(|nostr_group_id: [u8; 32]| {
+                self.storage()
+                    .find_group_by_nostr_group_id(&nostr_group_id)
+                    .ok()
+                    .flatten()
+                    .map(|group| group.mls_group_id)
+                    .or_else(|| Some(GroupId::from_slice(&nostr_group_id)))
+            })
+            .unwrap_or_else(|| GroupId::from_slice(&[0u8; 32]))
```
Comment (erskingardner) [inline crates/mdk-core/src/messages.rs]: Why return a 0 `GroupId` here? That feels sort of strange to me.

**Review by jgmontoya — APPROVED** [review]
